### PR TITLE
Add research progress to nations view research column

### LINKED ISSUE #1767: Include current progress next to research name in nations view
**Is your feature request related to a problem? Please describe.**
not related to a problem

**Describe the solution you'd like**
Currently, the column "Research" only says the name of the research target (e.g. "Pottery"). To know the current progress, one must press the nation and check the detailed information at the bottom of the client. The research column could include the current progress (i.e. "Pottery (3/30)"), becoming essentially identical to the information displayed in the details.

### PR DESCRIPTION
**Pull request by jwrober**
Closes #1767

### CODE REVIEW

File: client/plrdlg_common.cpp [hunk 1]
```diff
@@ -436,8 +436,9 @@ QString col_research(const struct player *them)
     return _("-");
   } else if (BV_ISSET(them->client.visible, NI_TECHS)) {
     struct research *research = research_get(them);
-    return research_advance_name_translation(research,
-                                             research->researching);
+    return research_advance_name_translation(research, research->researching)
+           + " " + "(" + QString::number(research->bulbs_researched) + "/"
+           + QString::number(research->client.researching_cost) + ")";
```
Comment (lmoureaux) [inline client/plrdlg_common.cpp:441]: Avoid building strings this way, it makes it harder to translate if needed later. Prefer the string-building function `arg`:
```suggestion
    return QStringLiteral("%1 (%2/%3)").arg(research_advance_name_translation(research, research->researching))
           .arg(research->bulbs_researched).arg(research->client.researching_cost);
```
(with proper `clang-format`ting)

**Review by lmoureaux — CHANGES_REQUESTED** [review]
Gives strange results when the target isn't set:
image

**Comment by jwrober** [comment]
So I got the translatable strings, not sure what to do with the other comment.

File: client/plrdlg_common.cpp [hunk 2]
```diff
@@ -436,13 +436,23 @@ QString col_research(const struct player *them)
     return _("-");
   } else if (BV_ISSET(them->client.visible, NI_TECHS)) {
     struct research *research = research_get(them);
-    return research_advance_name_translation(research,
-                                             research->researching);
+    if (research->client.researching_cost == 0) {
+      // Nothing is selected to research
+      return QStringLiteral("%1").arg(research_advance_name_translation(
```
Comment (lmoureaux) [inline client/plrdlg_common.cpp:441]: Maybe `%1 (%2/-)`? This way one sees how many bulbs are stored

**Comment by lmoureaux** [comment]
> So I got the translatable strings, not sure what to do with the other comment.

I don't see any other comment, so fine :)

File: client/plrdlg_common.cpp (hunk 2 above)
Comment (jwrober) [reply client/plrdlg_common.cpp:441]: Ok I'll push another update. Good idea.

**Review by lmoureaux — APPROVED** [review]

**Pull request merged**